htdocs/kernel/module.php

```
@@ -326,6 +326,21 @@ public function loadInfo($dirname, $verbose = true)
         }
         include $file;
+        // Keep the manifest's own spelling of its dirname only when it names THE SAME
+        // FOLDER this was loaded from, compared case-insensitively -- that is what a
+        // case-insensitive filesystem treats as one directory, and a module's own casing
+        // is worth preserving. A manifest that declares a different, traversal, or empty
+        // dirname is not trusted to redirect anything: fall back to the real folder
+        // ($dirname, already basename()'d above). This blocks a hostile package from
+        // poisoning every getInfo('dirname') consumer and, via loadInfoAsVar(), the
+        // persisted xoops_modules.dirname column, or from pointing metadata loads at
+        // another module's directory.
+        if (isset($modversion) && is_array($modversion)) {
```

PHPMD (2.15.0)

[warning] Avoid using undefined variables such as '$modversion' which will lead to PHP notices. (undefined)

(UndefinedVariable)